### PR TITLE
FE: Reflect to default default value in action mapping form

### PR DESCRIPTION
**Pull request by uladzimirdev**
Epic: https://github.com/metabase/metabase/issues/30016

### Description

Based on discussion [in slack](https://metaboat.slack.com/archives/C04FG88HL95/p1686576647410289?thread_ts=1686051160.972209&cid=C04FG88HL95) we need to allow user to submit form if default value is available for parameter in mapping form

### CODE REVIEW

**Review by ranquild — APPROVED** [review]
@uladzimirdev I don't see a test that the user can submit the mapping form when the parameter has the default value (new condition in this PR), could you double check?

**Comment by uladzimirdev** [comment]
> @uladzimirdev I don't see a test that the user can submit the mapping form when the parameter has the default value (new condition in this PR), could you double check?

added, thanks

File: frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx [hunk 1]
```diff
@@ -85,19 +92,20 @@ export const ActionParameterMappingForm = ({
...
+                  name: defaultValue,
                   value: null,
                 },
                 ...dashboardParameters.map(dashboardParam => ({
                   key: dashboardParam.id,
                   name: dashboardParam.name,
                   value: dashboardParam.id,
                 })),
-              ]}
+              ].filter(Boolean)}
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx:108]: I like your point, but it's a pretty common practice to use `filter(Boolean)`

File: frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx [hunk 2]
```diff
@@ -64,10 +68,13 @@ export const ActionParameterMappingForm = ({
       {actionParameters.map((actionParam: WritebackParameter) => {
         const isHidden = isParameterHidden(action, actionParam);
         const isRequired = isParameterRequired(action, actionParam);
+        const defaultValue = getParameterDefaultValue(action, actionParam);
         const mappedValue = currentMappings[getTargetKey(actionParam)];
         const isParameterMapped = mappedValue != null;
+        const hasDefaultValue = defaultValue != null;
 
-        const showError = isHidden && isRequired && !isParameterMapped;
+        const showError =
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx:76]: you're right, but it's only 2 places at the moment. can be abstracted later 😄 there should be more changes related to default value anyway in the followup

File: frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx [hunk 3]
```diff
@@ -98,73 +100,214 @@ const setup = (
...
-    it("allows to submit a form", () => {
-      expect(screen.getByRole("button", { name: "Done" })).toBeEnabled();
-    });
-  });
+        expect(within(popover).getByText(DEFAULT_VALUE)).toBeInTheDocument();
+        expect(
+          within(popover).queryByText("Select a value"),
+        ).not.toBeInTheDocument();
+        expect(
+          within(popover).queryByText("Ask the user"),
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx:267]: goot catch

File: frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kamilmielnik) [reply frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx:108]: It is a common practice to use `filter(Boolean)` but I don't think Kelvin's comment was about that. 
I agree that it's a bit too hard to read and I like Kelvin's suggestion to declamp this.

File: frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/actions/components/ActionViz/ActionDashcardSettings.unit.spec.tsx:312]: added

File: frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/actions/components/ActionViz/ActionParameterMapper.tsx]: moved 😄

**Pull request merged**